drivers/pn532/pn532.c

```diff
@@ -0,0 +1,639 @@
+/*
+ * Copyright (C) 2016 TriaGnoSys GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup  drivers_pn532
+ *
+ * @{
+ * @file
+ * @brief   PN532 driver
+ *
+ * @author  Víctor Ariño <victor.arino@triagnosys.com>
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include "xtimer.h"
+#include "mutex.h"
+#include "pn532.h"
+#include "periph/gpio.h"
+#include "periph/i2c.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#define PN532_I2C_ADDRESS             (0x24)
+
+/* Commands */
+#define CMD_FIRMWARE_VERSION          (0x02)
+#define CMD_READ_REG                  (0x06)
+#define CMD_WRITE_REG                 (0x08)
+#define CMD_READ_GPIO                 (0x0c)
+#define CMD_WRITE_GPIO                (0x0E)
+#define CMD_SAM_CONFIG                (0x14)
+#define CMD_RF_CONFIG                 (0x32)
+#define CMD_DATA_EXCHANGE             (0x40)
+#define CMD_DESELECT                  (0x44)
+#define CMD_LIST_PASSIVE              (0x4a)
+#define CMD_RELEASE                   (0x52)
+
+/* Mifare specific commands */
+#define MIFARE_CMD_READ               (0x30)
+#define MIFARE_CMD_WRITE              (0xA0)
+
+/* RF register settings */
+#define RF_CONFIG_MAX_RETRIES         (0x05)
+
+/* Buffer operations */
+#define BUFF_CMD_START                (6)
+#define BUFF_DATA_START               (BUFF_CMD_START + 1)
+#define RAPDU_DATA_BEGIN              (1)
+#define RAPDU_MAX_DATA_LEN            (PN532_BUFFER_LEN - BUFF_DATA_START - 5)
+#define CAPDU_MAX_DATA_LEN            (PN532_BUFFER_LEN - BUFF_DATA_START - 1)
+
+/* Length for passive listings */
+#define LIST_PASSIVE_LEN_14443(num)   (num * 20)
+
+#if ENABLE_DEBUG
+#define PRINTBUFF printbuff
+static void printbuff(char *buff, unsigned len)
+{
+    while (len) {
+        len--;
+        printf("%02x ", *buff++);
+    }
+    printf("\n");
+}
+#else
+#define PRINTBUFF(...)
+#endif
+
+static void _nfc_event(void *dev)
+{
+    mutex_unlock(&((pn532_t *)dev)->trap);
+}
+
+void pn532_reset(pn532_t *dev)
+{
+    DEBUG("pn532: reset\n");
+    gpio_clear(dev->conf->reset);
+    xtimer_usleep(400000);
```

Comment:
Mostly ~~arbitrary~~ experimental: the pin is required to be low for a correct reset. Same below (L:89), if the pn532 is operated right after the reset it does not respond as expected. 

It's a good point, I didn't try lower values for this one (I did for the one in L:89), maybe @thomaseichinger can try it if he has an available board and time within the upcoming days.